eng/Directory.Build.Data.targets

```diff
@@ -33,12 +33,20 @@
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0-alpha-004" PrivateAssets="all" IsImplicitlyDefined="true" />
   </ItemGroup>
 
+  <!-- If signing is enabled for an SDK library, delay sign using the shared key -->
   <PropertyGroup Condition="'$(SignAssembly)' != 'false'">
     <SignAssembly>true</SignAssembly>
     <DelaySign>true</DelaySign>
     <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
 
+  <!-- If signing is enabled for a Test library, sign immediately using the test key -->
+  <PropertyGroup Condition="'$(SignTestAssembly)' == 'true' and '$(IsTestProject)' == 'true'">
```

Comment:
Yes this is what I was wanting. Not particularly important but I don't think we need the IsTestProject property in the condition.

Reply:
I believe that we do if the properties are being set at the level of the solution's Directory.Build.props and not in the individual project file(s) - which is my current implementation.  I'd like to keep that, as it feels like it should work from either level.

 I experimented with pulling these into the existing `IsTestProject` conditional above, but had to tag each with the `SignTestAssembly` condition due to the inability to nest `PropertyGroup` elements.  I also experimented with another block for "if this isn't a test assembly force the `SignTestAssembly` to false.

I'll admit that I'm not thrilled with the compound condition, but it seemed the lesser of evils compared against the other experiments.  At least with this approach the intent is clear at a glance instead of having some implicit resets floating in the environment that are potentially non-obvious.

Reply:
The case I had in mind was the IsTestLibraryHelper that might need to be signed but we can handle that if and when the case ever occurs. For now I'm fine with what you have.

Reply:
I missed that case.   Given that, I think it may be better to go with forced unset of not a test or helper project.   I'm going to throw that in.